pkg/csi/csi.go

```diff
@@ -60,13 +60,25 @@ func InitializeSidecars(ver string) error {
 
 	sv := version.MustParseGeneric(ver)
 
+	oneLessMinorImages := new(SidecarImages)
 	for _, image := range si {
 		v := version.MustParseGeneric(image.Version)
-		if sv.Major() == v.Major() && sv.Minor() == v.Minor() {
-			sidecarImages = &image
-			break
+		if sv.Major() == v.Major() {
+			if sv.Minor() == v.Minor() {
+				sidecarImages = &image
+				break
+			}
+			// for a brief period of time platform gets ahead of operator and we
+			// need to use image versions that are lagging one minor version of platform
+			// as next best option
+			if sv.Minor()-1 == v.Minor() {
+				oneLessMinorImages = &image
+			}
```

Comment:
`else` is always preferable to `break` for code structure and there is no need to use a new variable and then just copy it back into the original one again later.

```suggestion
		if sv.Major() == v.Major() {
			if sv.Minor() == v.Minor() {
				sidecarImages = &image

			// for a brief period of time platform gets ahead of operator and we
			// need to use image versions that are lagging one minor version of platform
			// as next best option	
			} else if sv.Minor()-1 == v.Minor() {
				sidecarImages = new(SidecarImages)
			}
```

Reply:
removed